### ISSUE TITLE
Is there a way to unset a key using environment variables?

### ISSUE THREAD

**Issue by grillorafael** [lightbend/config#582]
If I have a key with a default value set but I want to be able to  unset  in case I have a given `ENV` variable present. Is there a way to do it ?

**Comment by ktoso** [comment]
Set it to an empty string?

**Comment by havocp** [comment]
TL;DR not really.

The tricky thing about env vars is that they always contain strings, which means "VAR=null" comes through as the string "null" rather than a null value. No great solution to this... env vars only go so far. A more flexible solution in the long run can be to use env vars to specify a `-Dconfig.file=$FILE` kind of thing, where `$FILE` can be different for each distinct configuration you have, while including another file with common configuration.

There was some discussion on an issue I can't find of allowing JSON/HOCON to appear in env vars somehow... I don't remember exactly what that idea was but I think maybe it was a certain env var like `CONFIG_OVERRIDES='{ "some.key" : null }'`   ... or it could be `-Dconfig.env=MY_HOCON_VARNAME` I don't know.  Anyway that's a feature idea that could help with this sort of thing, many details to be worked out though.

**Comment by grillorafael** [comment]
Thanks for the answer @havocp and @ktoso
